### DISCUSSION TITLE
Dynamic day of week for startdate

### DISCUSSION THREAD

**Discussion by oulondo** [add2cal/add-to-calendar-button#591] [Q&A] [answered]
This app is amazing and so useful.

I want to create a recurring event every Thursday for the next 4 Thursdays. Is there a way to set the startDate property as "the next Thursday"?

I tried startDate="today" with the below code, hoping that it might start from today and find the next Thursday due to the recurrences RRULE but it didn't work.

<add-to-calendar-button
  name="Self-Defined Living"
  options="'Apple','Google','iCal','Microsoft365','MicrosoftTeams','Outlook.com','Yahoo'"
  location="SelfDefinedLiving.com/meetingroom"
recurrence="RRULE:FREQ=WEEKLY;INTERVAL=1;BYDAY=TH;COUNT=4"
  startTime="16:00"
  endTime="18:00"
  timeZone="US/Eastern" label="Add weekly calls to your calendar with this button" 
></add-to-calendar-button>

**Comment by jekuer** [comment]
Hi @oulondo,

Interesting case. 
This is not possible at the moment. Might consider it for the future.
It also feels very edgy. However, we need something similar to get another planned feature working (making "until" play nicely for recurrence).

For now: Imho, you don't need it. If it should always start at the next possible Thursday, you could simply set last Thursday. From a user's perspective it should not make that much of a difference.

**Reply by oulondo** [reply]
So if I use an arbitrary past Thursday as the start date. Then say the recurrence should be every Thursday for 4 recurrences. That should work by starting on the next available Thursday?

**Reply by jekuer** [reply, marked as answer]
ah, no. When you limit the number of occurrences, this would not work. Sorry.
For indefinite recurrence, it could work this way.

Then, for now, your only chance is to calculate the next day within your application and pass this date to the button.

Will put this on our roadmap.

**Reply by oulondo** [reply]
Ok, thanks for getting back to me.

**Comment by isacolick** [comment]
I am also running into this issue where I am setting up an indefinite weekly occurrence for a Friday meeting. If use startDate="today" it ignores the RRULE's BYDAY setting and uses today day of week. It looks like I have to set the startDate to the upcoming Friday and change it every week, which isn't great. Is there a better option? Thank you!
